presto-main/src/main/java/com/facebook/presto/execution/QueryStateMachine.java

```diff
@@ -715,14 +719,15 @@ public boolean transitionToFinishing()
 
         Optional<TransactionInfo> transaction = session.getTransactionId()
                 .flatMap(transactionManager::getOptionalTransactionInfo);
+
         if (transaction.isPresent() && transaction.get().isAutoCommitContext()) {
             ListenableFuture<?> commitFuture = transactionManager.asyncCommit(transaction.get().getTransactionId());
             Futures.addCallback(commitFuture, new FutureCallback<Object>()
             {
                 @Override
-                public void onSuccess(@Nullable Object result)
+                public void onSuccess(@Nullable Object commitHandle)
                 {
-                    transitionToFinished();
```

Comment:
```java
if (result instanceof ConnectorCommitHandle) {
    this.commitHandle.set((ConnectorCommitHandle) commitHandle);
}

Reply:
I mean let's move the code logic here and do not change the interface for `transitionToFinished`.